### DISCUSSION TITLE
HLS over HTTPS

### DISCUSSION THREAD

**Discussion by technicaldesign** [bluenviron/mediamtx#502] [Questions] [answered]
Hi,

while experimenting with this great software, I found out I had problems with CORS. This is not because of cross-site issues, but because of a localhost HTTP request from an HTTPS host, which is hopelessly blocked by all browsers. My question is: will RTSP Simple Server ever support HLS over HTTPS directly, or perhaps am I missing something and it already does and I'm not understanding the protocol and/or the implementation?

Currently I'm reverse proxying the HTTP <> HTTPS with Apache (not a big deal but could be complicated for inexperienced users) and I was just wondering, since this software is such a simple and elegant solution and it already supports RTSPS and therefore manages certificates and keys (players apparently don't implement CORS). I suspect that most if not all HLS consumption is done through browsers so this - tricky and confusing with prefetch and specific headers - CORS issue is something to, perhaps, focus a bit on.

Thank you for your extremely elegant and functional work!

PS apologies and please ignore if it is already supported and I couldn't figure it out, otherwise, perhaps, treat ii as a feature request.

**Comment by aler9** [comment, marked as answer]
> will RTSP Simple Server ever support HLS over HTTPS directly

Hello, common HTTP servers like Nginx, Caddy or Apache provide logs, plugins, IP filtering, rate limiting and a lot of security features, that rtsp-simple-server doesn't provide, since it's not primarily an HTTP server. In my opinion it's better to use these in front of rtsp-simple-server if you want to expose the service to the internet.

**Comment by technicaldesign** [comment]
Makes a lot of sense, at some point IMHO RTSPSS should enter kind of a feature freeze (or at least slow down), otherwise there could be a risk of losing the "Simple" feature. At that stage, a very simple out-of-the box https implementation might be attractive for low-traffic/non-critical usage cases but In the end I agree that whoever needs foolproof CORS support is probably already self-hosting his/her's website which includes a player and therefore should be capable of reverse-proxying RTSPSS. It's just that CORS can be tricky to troubleshoot and understand since it's not very well covered at a beginner's level. Once more congrats and thanks for your great work.

Ciao from Rome

Stefano

PS (just realized at a very late stage you're Italian and kudos for your other great projects)

[a comment by an automated account is not shown]
